joomla/joomla-cms
# [4.0][DRAFT] replace hard coded icon by jlayout

Pull Request for Issue # .

### Summary of Changes

With this PR merged all hard coded icons `<span class="fas fa-some-kind-of-icon" aria-hidden="true" and-more-attributes></span>` are being replaced by the JLayout implemented with #30470 
The reason behind this is to make Joomla more flexible to switch to other icon libraries, or to implement your own. 
With this PR merged the declaration of icons is only be done in https://github.com/joomla/joomla-cms/blob/4.0-dev/layouts/joomla/icon/iconclass.php

### Testing Instructions

- Open Joomla Administrator and inspect icons
- Apply PR and refresh page
- All icons should be shown as it was before this PR. 

### Actual result BEFORE applying this Pull Request

All icons are hard coded.

### Expected result AFTER applying this Pull Request

Most of all icons are set through the `LayoutHelper('joomla.icon.iconclass', ['icon' => 'some-kind-of-icon-type'])`


### Documentation Changes Required

I don't know if there is documentation about the present LayoutHelpers and how they are / should be used.
